### PR TITLE
Ik 104

### PR DESCRIPTION
**Pull request by adrielFab**
Design of routes page is accomplished

### CODE REVIEW

File: client/app/src/main/java/com/example/mrides/Activity/RouteActivity.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+        CustomList customList = new CustomList(this, names, desc, imageid);
+
+        listView = (ListView) findViewById(R.id.listView);
+        listView.setAdapter(customList);
+
+        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            @Override
+            public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {
+                Toast.makeText(getApplicationContext(),"You Clicked "+names[i],Toast.LENGTH_SHORT).show();
```
Comment (handriamanantena) [inline client/app/src/main/java/com/example/mrides/Activity/RouteActivity.java]: You Clicked should be in the resource file.

File: client/app/src/main/java/com/example/mrides/Activity/RouteActivity.java [hunk 2]
```diff
@@ -1,31 +1,40 @@
...
         listView = (ListView) findViewById(R.id.listView);
         listView.setAdapter(customList);
 
     }
+
+    @Override
+    public void onClick(View view) {
+        Intent intent = new Intent(RouteActivity.this, HomePage.class);
+        startActivity(intent);
+        Toast.makeText(RouteActivity.this, "Ratings have been submitted", Toast.LENGTH_SHORT).show();
```
Comment (handriamanantena) [inline client/app/src/main/java/com/example/mrides/Activity/RouteActivity.java]: Ratings have been submitted need to be added in resource file.

File: client/app/src/main/java/com/example/mrides/CustomList.java [hunk 3]
```diff
@@ -14,16 +14,14 @@
 public class CustomList extends ArrayAdapter {
 
     private String[] names;
-    private String[] desc;
     private Integer[] imageid;
     private Activity context;
     private RatingBar ratingBar;
 
-    public CustomList(Activity context, String[] names, String[] desc, Integer[] imageid) {
+    public CustomList(Activity context, String[] names, Integer[] imageid) {
```
Comment (handriamanantena) [inline client/app/src/main/java/com/example/mrides/CustomList.java]: change variable names to username, or firstname.

File: client/app/src/main/java/com/example/mrides/Activity/RouteActivity.java [hunk 4]
```diff
@@ -33,8 +41,39 @@ protected void onCreate(Bundle savedInstanceState) {
...
+            default:
+                backToHome();
+                Toast.makeText(RouteActivity.this, "Ratings have been submitted", Toast.LENGTH_SHORT).show();
+                break;
+        }
+
+    }
+
+    public void promptUserCancellation() {
+        new AlertDialog.Builder(this)
```
Comment (handriamanantena) [inline client/app/src/main/java/com/example/mrides/Activity/RouteActivity.java:57]: remove this anonymous class. Should be a separate class or an interface implemented by the 
activity.

File: client/app/src/main/java/com/example/mrides/Activity/RouteActivity.java [hunk 5]
```diff
@@ -33,8 +41,39 @@ protected void onCreate(Bundle savedInstanceState) {
...
+                .setTitle("Delete route")
```
Comment (handriamanantena) [inline client/app/src/main/java/com/example/mrides/Activity/RouteActivity.java]: text should be in resource file.

File: client/app/src/main/java/com/example/mrides/Activity/RouteActivity.java [hunk 6]
```diff
@@ -33,8 +41,39 @@ protected void onCreate(Bundle savedInstanceState) {
...
+                .setMessage("Are you sure you want to delete this route?")
```
Comment (handriamanantena) [inline client/app/src/main/java/com/example/mrides/Activity/RouteActivity.java]: text should be in the resource file.

File: client/app/src/main/java/com/example/mrides/Activity/RouteActivity.java [hunk 7]
```diff
@@ -33,8 +41,39 @@ protected void onCreate(Bundle savedInstanceState) {
...
+                .setPositiveButton(android.R.string.yes, new DialogInterface.OnClickListener() {
+                    public void onClick(DialogInterface dialog, int which) {
+                        backToHome();
+                        Toast.makeText(RouteActivity.this, "Route deleted", Toast.LENGTH_SHORT).show();
```
Comment (handriamanantena) [inline client/app/src/main/java/com/example/mrides/Activity/RouteActivity.java]: add text in resource file.

File: client/app/src/main/java/com/example/mrides/Activity/RouteActivity.java [hunk 8]
```diff
@@ -33,8 +41,39 @@ protected void onCreate(Bundle savedInstanceState) {
...
+                    }
+                })
+                .setNegativeButton(android.R.string.no, new DialogInterface.OnClickListener() {
```
Comment (handriamanantena) [inline client/app/src/main/java/com/example/mrides/Activity/RouteActivity.java]: remove anonymous class.

**Review by handriamanantena — CHANGES_REQUESTED** [review]

File: client/app/src/main/java/com/example/mrides/Activity/RouteActivity.java (hunk 4 above)
Comment (adrielFab) [inline client/app/src/main/java/com/example/mrides/Activity/RouteActivity.java]: This is an alert dialog object

**Review by handriamanantena — APPROVED** [review]

**Pull request merged**